tests/acceptance/features/bootstrap/Provisioning.php

```diff
@@ -3510,96 +3502,14 @@ public function deleteGroup(string $group):void {
 	 * @throws GuzzleException
 	 */
 	public function groupHasBeenDeleted(string $group):void {
-		if (OcisHelper::isTestingWithGraphApi()) {
-			$this->graphContext->adminHasDeletedGroupUsingTheGraphApi($group);
+		if ($this->groupExists($group) && $this->isTestingWithLdap()) {
+			$this->deleteLdapGroup($group);
 		} else {
-			$this->deleteGroup($group);
+			$this->graphContext->adminHasDeletedGroupUsingTheGraphApi($group);
```

Comment:
I found that there are many functions in our code used for group deletion. Make sure that there's one function that deletes the group by calling helper and the rest function can call directly to that. 
For example: `adminDeletesGroupWithGroupId` looks like the function of a given statement as the status code is checked inside it. Try to use that and delete `adminDeletesGroupUsingTheGraphApi` as it's only used once. Check if there are multiple repeated step like this.

Reply:
@S-Panta Ta Yes, you are right. I was unaware of that. 
I made a few chanes. Would you please check if it's okay?